### PR TITLE
Add galaxy-{containers,tool-util,util} and zipstream-new

### PR DESCRIPTION
**Pull request by mr-c**
Revival of #12597

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mr-c** [comment]
Dear @croth1 , @mbargull , and @bgruening : do you each consent to be the maintainers of `galaxy-{containers,tool-util,util}` and `zipstream-new`?

File: recipes/galaxy-containers/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - galaxy.containers
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://galaxyproject.org
+  license: AFL-3.0
+  license_file: LICENSE.txt
```
Comment (nsoranzo) [inline recipes/galaxy-containers/meta.yaml:43]:
```suggestion
  license_file: LICENSE
```

which points to the file included in the source tar.gz archive. So you can remove `recipes/galaxy-containers/LICENSE.txt`

Same goes for the other galaxy-* packages.

**Comment by mr-c** [comment]
Thank you @nsoranzo ; would you like to be a listed maintainer?

**Comment by nsoranzo** [comment]
> Thank you @nsoranzo ; would you like to be a listed maintainer?

OK, for the galaxy-* ones

**Comment by mr-c** [comment]
This is ready for review & merging @bgruening

File: recipes/galaxy-containers/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
+    # Depend on galaxy-util during build time to avoid packaging galaxy/__init__.py .
+    - galaxy-util >={{ galaxy_version }}
+  run:
+    - galaxy-util >={{ galaxy_version }}
+    - python >=3.5
```
Comment (nsoranzo) [inline recipes/galaxy-containers/meta.yaml:26]:
```suggestion
    - python >=3.6
```

File: recipes/galaxy-containers/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
```
Comment (nsoranzo) [inline recipes/galaxy-containers/meta.yaml:21]:
```suggestion
    - python >=3.6
```

File: recipes/galaxy-tool-util/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    - mulled-build-tool=galaxy.tool_util.deps.mulled.mulled_build_tool:main
+    - mulled-build-files=galaxy.tool_util.deps.mulled.mulled_build_files:main
+    - mulled-list=galaxy.tool_util.deps.mulled.mulled_list:main
+    - mulled-update-singularity-containers=galaxy.tool_util.deps.mulled.mulled_update_singularity_containers:main
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
```
Comment (nsoranzo) [inline recipes/galaxy-tool-util/meta.yaml:30]:
```suggestion
    - python >=3.6
```

File: recipes/galaxy-tool-util/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    # Depend on galaxy-util during build time to avoid packaging galaxy/__init__.py .
+    - galaxy-util >={{ galaxy_version }}
+    # Depend on galaxy-containers due to shortcomings of bioconda-utils' build graph generation.
+    - galaxy-containers
+  run:
+    - galaxy-containers >=20.5.0
+    - galaxy-util >={{ galaxy_version }}
+    - python >=3.5
+    - six
```
Comment (nsoranzo) [inline recipes/galaxy-tool-util/meta.yaml:39]:
```suggestion
```

File: recipes/galaxy-util/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.5
```
Comment (nsoranzo) [inline recipes/galaxy-util/meta.yaml:20]:
```suggestion
    - python >=3.6
```

File: recipes/galaxy-util/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  run:
+    - bleach
+    - boltons
+    - docutils
+    - markupsafe
+    - packaging
+    - python >=3.5
```
Comment (nsoranzo) [inline recipes/galaxy-util/meta.yaml:27]:
```suggestion
    - python >=3.6
```

File: recipes/galaxy-util/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    - pyyaml
+    - requests
+    - routes
+    - six >=1.9.0
```
Comment (nsoranzo) [inline recipes/galaxy-util/meta.yaml:31]:
```suggestion
```

File: recipes/galaxy-containers/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  license: AFL-3.0
+  license_file: LICENSE
+  summary: Galaxy Container Modeling and Interaction Abstractions
+  doc_url: https://docs.galaxyproject.org
+  dev_url: https://github.com/galaxyproject/galaxy
+
+extra:
+  identifiers:
+    - biotools:galaxy
+    - doi:10.1093/nar/gky379
```
Comment (nsoranzo) [inline recipes/galaxy-containers/meta.yaml:51]: This section is bioconda-specific, IIRC.
```suggestion
```

File: recipes/galaxy-tool-util/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  license: AFL-3.0
+  license_file: LICENSE
+  summary: Galaxy Tool and Tool Dependency Utilities
+  doc_url: https://docs.galaxyproject.org
+  dev_url: https://github.com/galaxyproject/galaxy
+
+extra:
+  identifiers:
+    - biotools:galaxy
+    - doi:10.1093/nar/gky379
```
Comment (nsoranzo) [inline recipes/galaxy-tool-util/meta.yaml:78]:
```suggestion
```

File: recipes/galaxy-util/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+  license: AFL-3.0
+  license_file: LICENSE
+  summary: Galaxy Generic Utilities
+  doc_url: https://docs.galaxyproject.org
+  dev_url: https://github.com/galaxyproject/galaxy
+
+extra:
+  identifiers:
+    - biotools:galaxy
+    - doi:10.1093/nar/gky379
```
Comment (nsoranzo) [inline recipes/galaxy-util/meta.yaml:59]:
```suggestion
```

File: recipes/galaxy-containers/meta.yaml (hunk 9 above)
Comment (mr-c) [reply recipes/galaxy-containers/meta.yaml:51]: :-(

Comment (nsoranzo) [reply recipes/galaxy-containers/meta.yaml]: I know! But the linter complains...

**Review by nsoranzo — APPROVED** [review]

File: recipes/galaxy-containers/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  host:
+    - pip
+    - python >=3.6
+    # Depend on galaxy-util during build time to avoid packaging galaxy/__init__.py .
+    - galaxy-util >={{ galaxy_version }}
+  run:
+    - galaxy-util >={{ galaxy_version }}
+    - python >=3.6
+    - requests
+    # extra: docker
```
Comment (bgruening) [inline recipes/galaxy-containers/meta.yaml:28]: This can be removed now?

File: recipes/galaxy-tool-util/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+    - mulled-update-singularity-containers=galaxy.tool_util.deps.mulled.mulled_update_singularity_containers:main
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+    # Depend on galaxy-util during build time to avoid packaging galaxy/__init__.py .
+    - galaxy-util >={{ galaxy_version }}
+    # Depend on galaxy-containers due to shortcomings of bioconda-utils' build graph generation.
```
Comment (bgruening) [inline recipes/galaxy-tool-util/meta.yaml:33]: This comment and maybe the dep can be removed now?

**Review by bgruening — COMMENTED** [review]
Yes fine to be maintainer. Two comments inline.

File: recipes/zipstream-new/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,41 @@
+{% set name = "zipstream-new" %}
```
Comment (bgruening) [inline recipes/zipstream-new/meta.yaml:1]: @mr-c do you know why this is called `zipstream-new` and not `python-zipstream`?

Comment (mr-c) [reply recipes/zipstream-new/meta.yaml:1]: I used https://github.com/conda-incubator/grayskull to make this recipe based upon https://pypi.org/project/zipstream-new/ ; so the name must have come from there

**Comment by mr-c** [comment]
@bgruening and @nsoranzo ; I've invited you both to commit to my fork at https://github.com/mr-c/staged-recipes ; feel free to push commits directly to https://github.com/mr-c/staged-recipes/tree/adds_galaxy_tool_util

File: recipes/zipstream-new/meta.yaml (hunk 14 above)
Comment (bgruening) [reply recipes/zipstream-new/meta.yaml:1]: I see ok!

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks!

**Comment by mr-c** [comment]
Thank you @bgruening !

**Comment by bgruening** [comment]
@mr-c can you please remove them from bioconda?

**Comment by mr-c** [comment]
> @mr-c can you please remove them from bioconda?

Sure. I've never done that before, just delete the recipes like https://github.com/bioconda/bioconda-recipes/pull/31430 or are there other/additional steps?

**Pull request merged**